src/GithubExample.tsx

```diff
@@ -0,0 +1,93 @@
+import { navigate, RouteComponentProps } from "@reach/router";
+import { Button, Layout, Skeleton, Avatar, Row } from "antd";
+import axios from "axios";
+import React from "react";
+import "./App.css";
+import SimpleList from "./SimpleList";
+
+interface GithubExampleProps {
+  setClientAccessToken: React.Dispatch<string>;
+  githubAccessToken: string;
+}
+
+const GithubExample: React.FC<RouteComponentProps<
+  GithubExampleProps
+>> = props => {
+  if (
+    props &&
+    props.location &&
+    props.location.search &&
+    props.githubAccessToken === ""
+  ) {
+    const paramArray = props.location.search.split("=");
+    if (paramArray[0] === "?code") {
+      axios({
+        method: "get",
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+          "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
+          "Access-Control-Allow-Headers": "Content-Type, Authorization"
+        },
+        url: `https://github-oauth-jr.herokuapp.com/submit_code/${paramArray[1]}`
+      }).then(response => {
+        const { data } = response;
+
+        if (data && data.access_token && props.setClientAccessToken) {
+          props.setClientAccessToken(data.access_token);
+          navigate("/github_crud_auth");
+        }
+      });
+      return (
+        <div style={{ margin: 16 }}>
+          <Skeleton />
+        </div>
+      );
+    }
+  }
+  if (props.githubAccessToken !== "") {
```

Comment:
Maybe we should move everything from here any below to a separate custom component class. By doing that we can we can just generalize the naming/props to reuse this entire class for the admin page and swap out the component class.

Reply:
Refactored into smaller components, now the main component for GithubExample looks like this:

```typescript
const GithubExample: React.FC<RouteComponentProps<GithubExampleProps>> 
= props => {
  // /github_crud_auth + access token
  if (props.githubAccessToken !== "") {
    return (
      <GithubList
        clearAccessToken={() => {
          if (props && props.setClientAccessToken) {
            props.setClientAccessToken("");
          }
        }}
      />
    );
  } else {
    // /github_crud_auth + no access token
    return <GithubGuestPage />;
  }
};
```